Add intentions doc (#43)
fredrikhgrelland/terraform-nomad-hive: README.md

3. [Usage](#usage)
   1. [Providers](#providers)
   2. [Intentions](#intentions)
4. [Inputs](#inputs)
5. [Outputs](#outputs)
6. [Modes](#modes)
7. [Example](#example)
    1. [Verifying setup](#verifying-setup)
        1. [Data example upload](#data-example-upload)
8. [Authors](#authors)
9. [License](#license)
10. [References](#references)

## Prerequisites
Please follow [this section in original template](https://github.com/fredrikhgrelland/vagrant-hashistack-template#install-prerequisites)

## Compatibility
|Software|OSS Version|Enterprise Version|
|:---|:---|:---|
|Terraform|0.13.1 or newer||
|Consul|1.8.3 or newer|1.8.3 or newer|
|Vault|1.5.2.1 or newer|1.5.2.1 or newer|
|Nomad|0.12.3 or newer|0.12.3 or newer|

## Requirements

### Required software
All software is provided and run with docker.
See the [Makefile](Makefile) for inspiration.

## Usage
The following command will run hive in the [example/standalone](example/standalone) folder.
```sh
make up
```

### Providers
- [Nomad](https://registry.terraform.io/providers/hashicorp/nomad/latest/docs)
- [Vault](https://registry.terraform.io/providers/hashicorp/vault/latest/docs)

### Intentions

> Service mesh [communication controlled by intentions](https://learn.hashicorp.com/tutorials/consul/get-started-service-networking#control-communication-with-intentions).

Reviewer: Can you make this a more descriptive sentence? I'm not sure what this means.

Author: This is straight forward IMHO.
```
Service mesh communication controlled by intentions. 
Intentions are required when consul acl is enabled and default_policy is deny. 
```

Did u mean part about `Service mesh` (WHAT is it? WHY? LINKS?) or smth else?

Reviewer: More like a build-up to why I would use that link.

Author: Is it better described in https://github.com/fredrikhgrelland/terraform-nomad-hive/pull/43/commits/cfe214db12c728da9c672076ed366b3cccbe9a2f ?

Reviewer 2: Just to chime in, I like the one you linked better, Nikita. It gives some small context.

> Intentions are required when [consul acl is enabled and consul_acl_default_policy is deny](https://learn.hashicorp.com/tutorials/consul/access-control-setup-production#enable-acls-on-the-agents).

Reviewer 2: ```suggestion
Intentions are required when [`consul_acl=true` and `consul_acl_default_policy=deny`](https://learn.hashicorp.com/tutorials/consul/access-control-setup-production#enable-acls-on-the-agents).
```

Author: This module can be used without vagrant-hashistack, so I would like to decouple from its context in the documentation. WDYT?

```suggestion
Intentions are required when [consul acl is enabled and default_policy is deny](https://learn.hashicorp.com/tutorials/consul/access-control-setup-production#enable-acls-on-the-agents).
```

Reviewer 2: Oh right, that is a very good point. Didn't think of that at all.

```

#### Data example upload

> 

Reviewer: ```suggestion
```

Author: I am not sure that I understand ur suggestion.

Reviewer 2: I think he just means to remove one line for formatting's sake

Author: ahhh :)
